syntax = "proto3";

package enginetobts;

option go_package = "github.com/acompany-develop/QuickMPC/proto/engine_to_bts";

import "common_types/common_types.proto";

service EngineToBts {
    rpc GetTriples(GetRequest) returns (GetTriplesResponse) {}
    rpc GetRandBits(GetRequest) returns (GetRandBitsResponse) {}
}

message GetRequest{
    uint32 job_id = 1;
    uint32 amount = 2;
    int64 request_id = 3;
}

message Triple{
    pb_common_types.BigIntByte a = 2;
    pb_common_types.BigIntByte b = 3;
    pb_common_types.BigIntByte c = 4;
}

message GetTriplesResponse{
    repeated Triple triples = 1;
}

message GetRandBitsResponse{
    repeated int64 randbits = 1;
}
